pkg/analyze/redhat/clusteroperator.go

```diff
@@ -42,9 +47,78 @@ func (_ ClusterOperatorAnalyzer) Analyze(ctx context.Context, obj *status.Object
 		return status.UnknownStatusWithError(obj, err)
 	}
 
-	return analyze.AggregateResult(obj, nil, conditions)
+	// cloud-controller-manager references itself in the related objects
+	// so this is to avoid endless loop
+	if obj.Name == "cloud-controller-manager" {
+		return analyze.AggregateResult(obj, nil, conditions)
+	}
+
+	relatedObjects, _, err := unstructured.NestedSlice(obj.Unstructured.Object, "status", "relatedObjects")
+	if err != nil {
+		// do not add any substatuses in case of error
+		return analyze.AggregateResult(obj, nil, conditions)
+	}
+
+	subStatuses := c.evaluateRelatedObjects(ctx, relatedObjects)
+	return analyze.AggregateResult(obj, subStatuses, conditions)
+}
+
+func (c *ClusterOperatorAnalyzer) evaluateRelatedObjects(ctx context.Context, relatedObjects []interface{}) []status.ObjectStatus {
+	var statuses []status.ObjectStatus
+	for _, relObjec := range relatedObjects {
+		relObjecMap, ok := relObjec.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		resource := relObjecMap["resource"].(string)
+		group := relObjecMap["group"].(string)
+
+		gr := schema.GroupResource{Group: group, Resource: resource}
+
+		if slices.Contains(analyze.Register.IgnoredKinds(), c.evaluator.ResourceToKind(gr).GroupKind()) {
```

Comment:
thought: would it make sense for the register to provide `IsIgnoredGK` method instead, and hide how exactly the ignored kinds are stored and matched?